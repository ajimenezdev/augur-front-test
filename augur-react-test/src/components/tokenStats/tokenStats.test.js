import React from "react";
import renderer from "react-test-renderer";
import { fireEvent, render, waitForElement } from "@testing-library/react";
import { act } from "@testing-library/react-hooks";
import "@testing-library/jest-dom/extend-expect";
import TokenStats from "./index";

it("renders without crashing", () => {
  const component = renderer.create(
    <TokenStats token="" tokenStats={{}} onForceRefresh={() => {}} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders with arguments all values properly", () => {
  act(() => {
    const { getByText, getByLabelText } = render(
      <TokenStats
        token="0x1985365e9f78359a9b6ad760e32412f4a445e862"
        tokenStats={{
          averageTx: 1230000000000000000,
          averageTxFetching: false,
          medianTx: 3210000000000000000,
          medianTxFetching: false,
          richestAcc: "0x1985365e9f78359a9b6ad760e32412f4a445e872",
          richestAccFetching: false,
          mostActiveAcc: "0x1985365e9f78359a9b6ad760e32412f4a445e863",
          mostActiveAccFetching: false
        }}
        onForceRefresh={() => {}}
      />
    );

    expect(getByText(/0x1985365e9f78359a9b6ad760e32412f4a445e862/i));
    expect(getByLabelText("token-average")).toHaveTextContent("1.23");
    expect(getByLabelText("token-median")).toHaveTextContent("3.21");
    expect(getByLabelText("token-richest")).toHaveTextContent(
      "0x1985365e9f78359a9b6ad760e32412f4a445e872"
    );
    expect(getByLabelText("token-mostActive")).toHaveTextContent(
      "0x1985365e9f78359a9b6ad760e32412f4a445e863"
    );
  });
});

it("execute callback when button is clicked", async () => {
  await act(async () => {
    const mockRefresh = jest.fn();
    const { getByLabelText, container } = render(
      <TokenStats
        token="0x1985365e9f78359a9b6ad760e32412f4a445e862"
        tokenStats={{
          averageTx: 1230000000000000000,
          averageTxFetching: false,
          medianTx: 3210000000000000000,
          medianTxFetching: false,
          richestAcc: "0x1985365e9f78359a9b6ad760e32412f4a445e872",
          richestAccFetching: false,
          mostActiveAcc: "0x1985365e9f78359a9b6ad760e32412f4a445e863",
          mostActiveAccFetching: false
        }}
        onForceRefresh={mockRefresh}
      />
    );

    const buttonRefresh = getByLabelText("refresh-tokenStats");
    fireEvent.click(buttonRefresh);

    expect(mockRefresh.mock.calls.length).toBe(1);
  });
});
