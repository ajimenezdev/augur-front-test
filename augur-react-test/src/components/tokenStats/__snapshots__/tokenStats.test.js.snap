// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiCard-root tokenStats MuiPaper-rounded"
>
  <div
    className="MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
      >
        Token Stats
      </span>
      <span
        className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        
      </span>
    </div>
    <div
      className="MuiCardHeader-action"
    >
      <button
        aria-label="refresh-tokenStats"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <span
            aria-hidden={true}
            className="material-icons MuiIcon-root accountStats-refreshIcon"
          >
            refresh
          </span>
        </span>
      </button>
    </div>
  </div>
  <div
    className="MuiCardContent-root tokenStats-content"
  >
    <div
      className="tokenStats-infoRow"
    >
      <span
        className="tokenStats-label"
      >
        Average Token Transaction:
      </span>
      <span
        aria-label="token-average"
        className="tokenStats-value"
      />
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      className="tokenStats-infoRow"
    >
      <span
        className="tokenStats-label"
      >
        Median Token Transaction:
      </span>
      <span
        aria-label="token-median"
        className="tokenStats-value"
      />
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      className="tokenStats-infoCol"
    >
      <span
        className="tokenStats-label"
      >
        Richest Token holder:
      </span>
      <span
        aria-label="token-richest"
        className="tokenStats-value"
      />
    </div>
    <hr
      className="MuiDivider-root"
    />
    <div
      className="tokenStats-infoCol"
    >
      <span
        className="tokenStats-label"
      >
        MostActive Token holder:
      </span>
      <span
        aria-label="token-mostActive"
        className="tokenStats-value"
      />
    </div>
  </div>
</div>
`;
